project-ideas/fulfillment-center-mgmt/approveTransferOrder.md

```diff
@@ -0,0 +1,67 @@
+# Approve Transfer Orders
+
+1. The process to approve transfer order and correctly update the Order Item Status to either ITEM_PENDING_FULFILLMENT or ITEM_PENDING_RECEIPT on the basis of OH.statusFlowId.
+2. In the core Approve TO service, for each TO, header status update will happen and if the next valid item status transition for its statusFLowId is ITEM_PENDING_FULFILLMENT, the items should be iterated and reservations should happen.
```

Comment:
The OrderItem for TO, cannot be exploded like Sales Order in OMS. We should consider writing inventory reservation service, like in OOTB OFBiz.

Reply:
Discussed with @dixitdeepak Sir, the new reservation service being written by him will take care of the reservations for the explode OFF case also (for TO). The new service can be used here whenever item status is being moved to pending fulfillment.